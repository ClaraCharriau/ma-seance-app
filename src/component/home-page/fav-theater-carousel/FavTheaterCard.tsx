import { Link } from 'react-router-dom';
import { Theater } from '../../../models/Theater';
import style from '../fav-theater-carousel/FavTheater.module.css';
import 'pure-react-carousel/dist/react-carousel.es.css';

interface FavTheaterCardProps {
    theater: Theater;
}

const FavTheaterCard = (props: FavTheaterCardProps) => {
    const { theater } = props;

    const IMG_PATH = '/assets/img/theaters';
    const IMG_SUFFIX = '.jpg';

    return (
        <Link to={`/theaters/${theater.id}/day-1`} state={{ theater }} className={style.card}>
            <img
                className={style.theaterCardImg}
                src={IMG_PATH + theater.imgPath + IMG_SUFFIX}
                alt={'photo de ' + theater.name}
            />
            <div className={style.theaterName}>
                <p>{theater.name}</p>
            </div>
        </Link>
    );
};

export default FavTheaterCard;
