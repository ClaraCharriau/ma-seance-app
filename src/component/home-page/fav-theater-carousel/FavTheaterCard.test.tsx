import { render } from '@testing-library/react';
import FavTheaterCard from './FavTheaterCard';
import { BrowserRouter } from 'react-router-dom';
import mockTheater from '../../../mocks/theaters/theaters-6.json';

describe('Fav Theater Card Component', () => {
    it('renders Fav Theater Card component', () => {
        const component = render(
            <BrowserRouter>
                <FavTheaterCard theater={mockTheater} />
            </BrowserRouter>
        );

        expect(component.getByText('UGC Normandie')).toBeInTheDocument();
        expect(component.container).toMatchSnapshot();
    });
});
