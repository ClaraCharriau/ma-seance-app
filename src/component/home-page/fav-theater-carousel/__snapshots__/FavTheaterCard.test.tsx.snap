// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fav Theater Card Component renders Fav Theater Card component 1`] = `
<div>
  <a
    class="card"
    href="/theaters/6"
  >
    <img
      alt="photo de UGC Normandie"
      class="theaterCardImg"
      src="/assets/img/theaters/ugc-normandie-paris.jpg"
    />
    <div
      class="theaterName"
    >
      <p>
        UGC Normandie
      </p>
    </div>
  </a>
</div>
`;
